common/src/main/java/whocraft/tardis_refined/common/blockentity/door/GlobalDoorBlockEntity.java

```diff
@@ -18,15 +18,21 @@ public GlobalDoorBlockEntity(BlockPos blockPos, BlockState blockState) {
     }
 
 
-    public void onRightClick(BlockState blockState, ITardisInternalDoor door) {
+    public void onRightClick(BlockState blockState, ITardisInternalDoor door, Player player) {
         if (getLevel() instanceof ServerLevel serverLevel) {
 
             // we know that in this instance the serverlevel has a capability.
             TardisLevelOperator.get(serverLevel).ifPresent(cap -> {
                 if (cap.getInternalDoor() != door) {
                     cap.setInternalDoor(door);
                 }
-                if (!cap.getControlManager().isInFlight()) {
+                if(player.isCrouching()) {
```

Comment:
Make sure this is checking that its in flight first. Doors shouldn't be intractable in flight.